app/controllers/settings_controller.rb

```diff
@@ -16,6 +16,14 @@ def update
     end
   end
 
+  def regen_api_key
+    if current_user.update_attributes(api_key: SecureRandom.hex(16))
```

Comment:
We have `#genarate_api_key` in the User model to help us with this.  We _could_ do something like:

`if current_user.generate_api_key.save`, that DRYs up the use of using SecureRandom(16) everywhere.